node.gyp

```diff
@@ -123,6 +166,102 @@
     ],
   },
 
+  'target_defaults': {
+    'defines': [
+      'NODE_ARCH="<(target_arch)"',
+      'NODE_PLATFORM="<(OS)"',
+      'NODE_WANT_INTERNALS=1',
+      # Warn when using deprecated V8 APIs.
+      'V8_DEPRECATION_WARNINGS=1',
+    ],
+    'conditions': [
+      [ 'OS=="win"', {
+        'sources': [
+          'src/backtrace_win32.cc',
+          'src/res/node.rc',
+        ],
+        'defines!': [
+          'NODE_PLATFORM="win"',
+        ],
+        'defines': [
+          'FD_SETSIZE=1024',
+          # we need to use node's preferred "win32" rather than gyp's preferred "win"
+          'NODE_PLATFORM="win32"',
+          '_UNICODE=1',
+        ],
+        'libraries': [ '-lpsapi.lib' ]
+      }, { # POSIX
+        'defines': [ '__POSIX__' ],
+        'sources': [ 'src/backtrace_posix.cc' ],
+      }],
+      [ 'OS=="mac"', {
+        # linking Corefoundation is needed since certain OSX debugging tools
+        # like Instruments require it for some features
+        'libraries': [ '-framework CoreFoundation' ],
+        'defines!': [
+          'NODE_PLATFORM="mac"',
+        ],
+        'defines': [
+          # we need to use node's preferred "darwin" rather than gyp's preferred "mac"
```

Comment:
Can you fix the long lines and the capitalization/punctuation while you're here?